mysql-test/mysql-test-run.pl

```diff
@@ -3827,6 +3827,7 @@ ($)
 {
   my $tinfo= shift;
 
+  $ENV{'MYSQL_CURRENT_TEST_DIR'} = dirname($tinfo->{'path'});
```

Comment:
Is this actually used?

Reply:
Yes, there are some parts in generated tests where there is no need to substitute any parameters. Therefore there is no need to use perl 'print' to generate these absolutely the same parts for each set of parameters. So I moved these parts in separate *.inc files and just inserted '--source path_to_inc_file'" in generated file. But I need to know full path_to_inc_file because the generated test will be executed in temp dir, I can't use relative path there. I could, for example, just use mysql-test directory and add predefined path inside of this directory. But I don't like this idea, the tests can be moved in another subdirectory, the suite can be renamed etc., that's why I decided to introduce new environment variable. That does not require big changes, just one line, so it's feasible.